### PR TITLE
Allow use http proxy to download pecl packages

### PR DESCRIPTION
**Pull request by mistermoper**
Hi.

I am working on a project where the development with these features:
- User and group different than 1000.
- Proxy network.

When I try to build this image I can't download pecl packages because the proxy is rejecting the request.
With this PR we use the http_proxy environment variable to set it to the PHP image. If it's empty it won't make any effect, so it should be compatible with environments without http proxy.

Please review, Thanks!

### CODE REVIEW

**Comment by csandanov** [comment]
Hi, I don't get it, how would you pass this env var? You didn't add any build args.

**Comment by mistermoper** [comment]
Sorry! You are right, for an unknown reason it worked in a specific environment but not in a fresh environment.

Setting build arg and ENV arg to Dockerfile.

**Comment by mistermoper** [comment]
Can you check it now @csandanov ?

**Comment by csandanov** [comment]
No need to add the arg to env vars since it used only during the build and not needed in the environment

**Comment by mistermoper** [comment]
Removed!

**Comment by csandanov** [comment]
Also:

- set proxy via `pecl` instead of `pear` because we don't really use PEAR here, only PECL
- set it only when it's not empty
- rename env var to `PECL_HTTP_PROXY`
- pass it as the build arg in Makefile with the default empty value

**Comment by mistermoper** [comment]
When I've tried to set pecl config-set http_proxy I received this error:
`+ pecl config-set http_proxy http://proxy:80
config-set (http_proxy, http://proxy:80, user) failed, channel pecl.php.net`

I've tried this building the image in different hosts and received this error.
This is happening to more [people](https://blog.flowl.info/2015/peclpear-behind-proxy-how-to/).

Also it's mentioned that 

> PEAR / PECL won’t use the system proxy, instead you have to set it manually.

So it's not possible change use pear config-set with pecl config-set for this case, and it works correctly

**Comment by mistermoper** [comment]
@csandanov can you review now? Applied these changes:

- set it only when it's not empty
- rename env var to PECL_HTTP_PROXY
- pass it as the build arg in Makefile with the default empty value

File: 7/Dockerfile [hunk 1]
```diff
@@ -152,7 +153,10 @@ RUN set -xe; \
       docker-php-ext-install "-j${NPROC}" gd; \
     \
     pecl config-set php_ini "${PHP_INI_DIR}/php.ini"; \
-    \
+    if [[ "${PECL_HTTP_PROXY}" != "" ]]; then \
```
Comment (csandanov) [inline 7/Dockerfile]: the correct way to check for emptiness in bash is using `-n`, see examples in this dockerfile

**Comment by mistermoper** [comment]
@csandanov corrected.

**Pull request merged**